### PR TITLE
Add standalone script compatibility

### PR DESCRIPTION
**Pull request by josh-hemphill**
#### Add UMD (Universal Module) webpacked file for inclusion in plain js projects and automatic inclusion in UNPKG CDN
This would have the added benefit of being able to create codepens etc. for examples and testing.
I also added a basic `example.html` which includes the UMD and lets you test/experiment with it.

##### Note: No modification to `index.js`, just added webpack dev-dependency, `npm run build` command, `example.html`, and UMD directory/file (maybe eventually add beautified UMD `expression-eval.js`  so CDN users can view source in codepens etc.)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,6 +32,11 @@
...
   },
   "devDependencies": {
-    "budo": "^11.6.3"
-  }
+    "budo": "^11.6.3",
+    "webpack": "^4.41.6",
+    "webpack-cli": "^3.3.11"
+  },
+	"engines": {
+		"node": ">= 6.0.0"
```
Comment (donmccurdy) [inline package.json]: What is the engine requirement based on?

**Comment by donmccurdy** [comment]
Hi! Unfortunately I don't use Webpack regularly, and would prefer not to take on maintaining a build artifact that requires it. Would [microbundle](https://github.com/developit/microbundle)'s `unpkg` output work for you? If so, I think this structure could be OK:

```
  "source": "index.js",                        // source file
  "main": "dist/expression-eval.js",           // output path for CommonJS/Node
  "module": "dist/expression-eval.module.js",  // output path for JS Modules
  "unpkg": "dist/expression-eval.umd.js",      // optional, for unpkg.com
```

It'd be fine to omit `example.html ` as well, I'm willing to trust the test suite and microbundle's maintainers.

**Comment by josh-hemphill** [comment]
Yeah, microbundle works great, I didn't know it existed.
And I guess I went overboard with the example.html, since having a expression-eval.umd.js is supposed to allow making codepens etc. which would would fill that purpose.
I'll make a push shortly.

**Comment by josh-hemphill** [comment]
NPM is reporting that microbundle has several security vulnerabilities.  So I added a notice not to use the standalone scripts in production at the moment.

File: package.json (hunk 1 above)
Comment (josh-hemphill) [reply package.json]: Webpack had some mention of it being sometimes necessary to specify. I think it was webpack's minimum supported version of node.

**Comment by RobertoMaurizzi** [comment]
Shouldn't those vulnerabilities only affect the `npm run build` step, i.e they won't affect the generated code? Modules should be safe (and in addition the only `High` one involves `svgo`, an SVG optimizer)

**Comment by donmccurdy** [comment]
Right, the vulnerabilities are essentially false positives, and would only affect the person publishing the package to npm with `devDependencies` in any case (i.e. me). Let's remove the warning, otherwise this looks good. Thanks!

**Comment by donmccurdy** [comment]
Thanks!

**Comment by josh-hemphill** [comment]
@donmccurdy I apologize, I don't think I tested it. Microbundle doesn't redefine `require()`. I think it's designed for front-end-only modules. If you're set on microbundle, one solution would be to `import jsep from 'jsep'` and '`export=` instead of `module.exports=`, but this would prevent older versions of node from using it. And additionally, using that way, imports all the functions directly under `window.exports` e.g. `window.exports.parse()`.

**Comment by donmccurdy** [comment]
It looks like Microbundle expects the input source code to use ES modules. It still creates CJS-compatible outputs from that. I've updated accordingly. For the standalone script use case, this should work:

```html
<script src="https://unpkg.com/jsep@0.3.4/build/jsep.js"></script>
<script src="https://unpkg.com/expression-eval@3.0.4/dist/expression-eval.umd.js"></script>
<script>

const fn = expressionEval.compile('a+3');
console.log('result: ', fn({a:4}));

</script>
```

**Comment by josh-hemphill** [comment]
Awesome, that works great.
BTW, I made a [codepen](https://codepen.io/josh-hemphill/pen/KKpgeJQ) that people could use to try out how it behaves with certain inputs. I ended up using something like it a lot to determine how to handle different edge cases and such.

**Pull request merged**